samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -98,8 +105,16 @@ private TestRunner() {
     configs.put(JobConfig.PROCESSOR_ID(), "1");
     configs.put(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY, PassthroughJobCoordinatorFactory.class.getName());
     configs.put(TaskConfig.GROUPER_FACTORY(), SingleContainerGrouperFactory.class.getName());
+    // Changing the base directory for non-changelog stores used by Samza application to separate the
+    // on-disk store locations for concurrently executing tests
+    configs.put(JobConfig.JOB_NON_LOGGED_STORE_BASE_DIR(),
+        FileSystems.getDefault().getPath("/tmp/" + this.inMemoryScope).toAbsolutePath().toString() + File.separator);
```

Comment:
Instead of hardcoding path, can you check if `Files.createTempDirectory()` works?

Reply:
Files.createTempDirectory() takes in the absolute path and involves manually deletion as well so this works better

Reply:
Are you sure? From the javadoc for the 2-arg variant:
`Creates a new directory in the default temporary-file directory, using the given prefix to generate its name.`

Reply:
I tested it, doc says it creates directory with the prefix in the default temporary directory, that might be not set to /tmp/

it generated for me in /private/var/folders/0p/sjzh971n0nzfnj3k3kjd7qvm000whj/T/sanil-jain1818152723896917306

I guess putting it in /tmp/ is a better option since we are aware where temp state directories get created

Reply:
Why does it matter? You can always log the directory path somewhere later.

Reply:
https://softwareengineering.stackexchange.com/questions/314796/should-temporary-files-be-saved-to-tmp-or-the-current-working-directory

Reply:
Advantages of using "OS temporary directory", which that /private/var/folders directory is on OSX.
https://apple.stackexchange.com/questions/94964/where-is-the-temp-folder/131080